include/navigation.hpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * @file navigation.hpp
+ * @author Hritvik Choudhari (hac@umd.edu)
+ * @brief Declaration file for the Navigation class, responsible for generating the search path and moving the robot.
+ * @version 0.1
+ * @date 2023-12-09
+ * 
+ * @copyright Copyright (c) 2023
+ * 
+ */
+#pragma once
+#include "rclcpp/rclcpp.hpp"
+#include "geometry_msgs/msg/pose.hpp"
+#include "./Perception.hpp"
+#include "geometry_msgs/msg/pose_stamped.hpp"
+#include "nav_msgs/msg/odometry.hpp"
+
+#include <memory>
+#include <iostream>
+#include <vector>
+#include <chrono>
+
+using POSE = geometry_msgs::msg::PoseStamped;
+using PUBLISHER = rclcpp::Publisher<geometry_msgs::msg::PoseStamped>::SharedPtr;
+using TIMER = rclcpp::TimerBase::SharedPtr;
+using ODOM = nav_msgs::msg::Odometry;
+using std::placeholders::_1;
+using std::chrono::duration;
+using namespace std::chrono_literals;
+
+/**
+ * @brief Navigation class responsible for generating the search path and moving the robot.
+ * 
+ * The Navigation class is designed to execute a search algorithm in the map, move the robot and book to the respective shelf,
+ * and resume the search algorithm after that. The Perception class is a friend class of Navigation.
+ */
+class Navigation : public rclcpp::Node {
+ public:
+    /**
+     * @brief Construct a new Navigation object.
+     * 
+     * Initializes the Navigation node and sets up necessary parameters.
+     */
+    Navigation();
+
```

Comment:
Add a destructor if creating a constructor manually. if default constructor is going to be used don't create one